libs/libxx/libcxx.cmake

```diff
@@ -18,10 +18,28 @@
 #
 # ##############################################################################
 
-if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/libcxx/.git)
+if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/libcxx)
 
   set(LIBCXX_VERSION 12.0.0)
 
+  set(LIBCXX_PATCH_COMMAND
+      patch -p0 -d ${CMAKE_CURRENT_LIST_DIR} <
+      ${CMAKE_CURRENT_LIST_DIR}/0001-Remove-the-locale-fallback-for-NuttX.patch
```

Comment:
@anchao 
there is a code-style problem here.
if cmake-format is used, the statement will be modified to :
```cmake
set(LIBCXX_PATCH_COMMAND
    patch
    -p0
    -d
    ${CMAKE_CURRENT_LIST_DIR}
    <
    ....
)
``` 
readability gets worse.
should we also use whitelist for cmake files?

Reply:
can we merge into one line?

Reply:
You can locally disable and enable code formatting by using the special comments

```
# cmake-format: off
code block
# cmake-format: on
```

https://cmake-format.readthedocs.io/en/latest/format-features.html

Reply:
> can we merge into one line?

i don't think it's good, if so, this line will be very long.

Reply:
> https://cmake-format.readthedocs.io/en/latest/format-features.html

thanks a lot, this is very useful